src/views/product-tutorials-view/helpers/detect-and-reformat-learn-url.ts

```diff
@@ -0,0 +1,69 @@
+import { ProductOption, ThemeOption } from 'lib/learn-client/types'
+import { getCollectionSlug } from 'views/collection-view/helpers'
+import { getTutorialSlug } from 'views/collection-view/helpers'
+import { getTutorial } from 'lib/learn-client/api/tutorial'
+import isAbsoluteUrl from 'lib/is-absolute-url'
+
+/**
+ * Given a URL string,  detect whether it is a collection or tutorial URL
+ * in the format used on learn.hashicorp.com.
+ *
+ * If it is, then reformat the URL to work with dev-dot's URL structure.
+ * Otherwise, return the URL unmodified.
+ */
+async function detectAndReformatLearnUrl(url: string): Promise<string> {
```

Comment:
I'm seeing that we need to create a generic handler for this. This work is reminiscent of the [rewriting links logic](https://github.com/hashicorp/dev-portal/blob/main/src/lib/remark-plugins/rewrite-tutorial-links/utils/handle-learn-links.ts) in the remark plugin. 

I wonder if you could lean on the [tutorials-map endpoint ](https://github.com/hashicorp/dev-portal/blob/main/src/pages/api/tutorials-map.ts)to get the slug...Since it's already cached. Not mandatory, I'm just seeing how we can reuse some code here if it makes sense to.

Reply:
Great call - updated to aim for this - leaning on `getTutorialsMap` & `rewriteTutorialsLink`, with some light refactoring in the latter to expose a more generic handler than can be called outside of remark plugin use.